relational_operators/DeleteOperator.cpp

```diff
@@ -81,19 +85,25 @@ void DeleteWorkOrder::execute(QueryContext *query_context,
 
   // TODO(harshad): Stream the block ID only if the predicate returned at least
   // one match in the StorageBlock.
-  ForemanMessage message(ForemanMessage::DataPipelineMessage(delete_operator_index_,
-                                                             input_block_id_,
-                                                             rel_id_));
+  serialization::DataPipelineMessage proto;
+  proto.set_operator_index(delete_operator_index_);
+  proto.set_block_id(input_block_id_);
+  proto.set_relation_id(rel_id_);
+
+  const std::size_t proto_length = proto.ByteSize();
+  char *proto_bytes = static_cast<char*>(std::malloc(proto_length));
```

Comment:
A couple of reasons:

 1. The length of `proto` is not fixed,
 2. This implementation is used in other places in the codebase as well.